pytest-backend/exasol/pytest_backend/itde_config.py

```diff
@@ -0,0 +1,236 @@
+        """Environment variable name"""
+
+        def normalize(name):
+            name = name.replace("-", "_")
+            name = name.upper()
+            return name
+
+        return f"{normalize(self.prefix)}_{normalize(self.name)}"
+
+    @property
+    def cli(self):
+        """Cli argument name"""
+
+        def normalize(name):
+            name = name.replace("_", "-")
+            name = name.lower()
+            return name
+
+        return f"--{normalize(self.prefix)}-{normalize(self.name)}"
+
+    @property
+    def pytest(self):
+        """Pytest option name"""
+
+        def normalize(name):
+            name = name.replace("-", "_")
+            name = name.lower()
+            return name
+
+        return f"{normalize(self.prefix)}_{normalize(self.name)}"
+
+    @property
+    def help(self):
+        """Help text including information about default value."""
+        if not self.default:
+            return f"{self.help_text}."
+        return f"{self.help_text} (default: {self.default})."
+
+
+class ItdeOptionGroup:
+    """
+    Wraps a set of pytest options.
+    """
+
+    def __init__(self, prefix, options):
+        self._prefix = prefix
+        self._options = tuple(ItdeOption(prefix=prefix, **kwargs) for kwargs in options)
+        self._default = {o.name: o.default for o in self._options}
+        self._env = {}
+        self._cli = {}
+        self._kwargs = ChainMap(self._cli, self._env, self._default)
+
+    @property
+    def prefix(self):
+        """The option group prefix."""
+        return self._prefix
+
+    @property
+    def options(self):
+        """A tuple of all options which are part of this group."""
+        return self._options
+
+    def kwargs(self, environment, cli_arguments):
+        """
+        Given the default values, the passed environment and cli arguments it will
+        take care of the prioritization for the option values in regard of their
+        source(s) and return a kwargs dictionary with all options and their
+        appropriate value.
+        """
+        env = {
+            o.name: o.type(environment[o.env])
+            for o in self._options
+            if o.env in environment
+        }
+        cli = {
+            o.name: getattr(cli_arguments, o.pytest)
+            for o in self.options
+            if hasattr(cli_arguments, o.pytest)
+            and getattr(cli_arguments, o.pytest) is not None
+        }
+        self._env.update(env)
+        self._cli.update(cli)
+        return self._kwargs
+
+
+@dataclass
+class OnpremDBConfig:
+    """Exasol database configuration"""
+
+    host: str
+    port: int
+    username: str
+    password: str
+
+
+@dataclass
+class OnpremBfsConfig:
+    """Bucketfs configuration"""
+
+    url: str
+    username: str
+    password: str
+
+
+@dataclass
+class SshConfig:
+    """SSH configuration"""
+
+    port: int
+
+
+@dataclass
+class ItdeConfig:
+    """Itde configuration settings"""
+
+    db_version: str
```

Comment:
Consider having a config module, to improve readability and naming.


```python
from ... import config

config = config.Itde
config = config.Ssh
config = config.OnPremDB
...
```
VS
```python
config = ItdeConfig
config = SshConfig
....
```

Reply:
This is how I understand the suggestion. Please correct me if I am wrong.

Wrap the dataclasses `OnpremDBConfig`, `OnpremBfsConfig`, `SshConfig` and `ItdeConfig` into an outer dataclass, which may be called for example `OnpremConfig`. In the `__init__.py` we will import just the `OnprmeConfig` instead of all that lot. Also, we can have just one fixture - `def onprem_config(request) -> OnpremConfig` instead of four. That would make sense to me.

Reply:
No, the "initial" idea was to rename the data classes (e.g., to `Ssh`, `Itde`, `OnPremDb`) and place them into a module called `config`. Then, you can import the `config` module and reference the configurations, resulting in shorter names and less repetition. Ultimately, something similar could be achieved by having an outer class. 
I think either approach would be better than having the semantic information repeated in the individual names.

Reply:
I see, sorry for the misunderstanding. I think the outer class is better because it also allows it to reduce the number of fixtures.